app/assets/javascripts/views/courses.js

```diff
@@ -101,5 +111,27 @@ Yacs.views.courses = function (data) {
       c.classList[isSelected ? 'remove' : 'add']('selected');
     });
     if (isCourseSelected(c)) c.classList.add('selected');
+
+    var desc = c.querySelector('course-description');
+    var showButton = c.querySelector('show-hide-button');
+    Yacs.on('click', showButton, function (showButton, event) {
+      var descTruncated = desc.classList.contains("truncated");
+      desc.classList[descTruncated ? 'remove' : 'add']("truncated");
+      showButton.classList[descTruncated ? 'add' : 'remove']("open");
```

Comment:
The button has it's own class because there are several style changes between when it's on "show" and "hide" that I figured would be cleaner to handle with a CSS class rather than use Javascript to change each individual style.